internal/ingress/controller/controller.go

```diff
@@ -573,6 +573,11 @@ func (n *NGINXController) getBackendServers(ingresses []*ingress.Ingress) ([]*in
 			}
 
 			for _, path := range rule.HTTP.Paths {
+				if path.Backend.Service == nil {
+					// skip non-service backends
+					continue
+				}
```

Comment:
Personally I do not oppose myself of doing a redirect to the default backend if user creates an ingress pointing to a non service backend.

I just wanted the Ingress to give a feedback to the user of why that  backend is being skipped, so if you can just add a log message would be great:

```suggestion
				if path.Backend.Service == nil {
					// skip non-service backends
					klog.V(3).Infof("Ingress %q and path %q does not contain a service backend, using default backend", ingKey, path.Path)

					continue
				}
```